test/e2e.test.ts

```diff
@@ -0,0 +1,279 @@
+    investorFundsTarget: deploymentParameters.investorFundsTarget.address,
+    gnoToken: gnoToken.address,
+    usdcToken: usdcToken.address,
+    wrappedNativeToken: wethToken.address,
+    communityFundsTarget: deploymentParameters.communityFundsTarget.address,
+    usdcPrice: deploymentParameters.usdcPrice,
+    gnoPrice: deploymentParameters.gnoPrice,
+    nativeTokenPrice: deploymentParameters.nativeTokenPrice,
+    teamController: deploymentParameters.teamController.address,
+  };
+  const vCowToken = await CowSwapVirtualToken.deploy(
+    ...constructorInput(ContractName.VirtualToken, deploymentParams),
+  );
+  const deploymentTimestamp = (await ethers.provider.getBlock("latest"))
+    .timestamp;
+
+  return {
+    cowToken,
+    vCowToken,
+    usdcToken,
+    gnoToken,
+    wethToken,
+    deploymentTimestamp,
+  };
+}
+
+describe("e2e-tests", () => {
+  // In the actual deployment cowDAO, communityFundsTarget, investorFundsTarget, teamController are all gnosis safes.
+  // But to keep the e2e tests simple, we use normal EOA.
+  // This should not make a difference, as in the e2e tests they are only used to send and receive tokens
+  const [
+    deployer,
+    user,
+    user_not_eligible,
+    cowDao,
+    communityFundsTarget,
+    investorFundsTarget,
+    teamController,
+  ] = waffle.provider.getWallets();
+  const initialCowSupply = ethers.utils.parseEther("1000");
+
+  // Prices are chosen to be realistic but yet easy to work with, so that the
+  // test amounts are meaningful.
+  // - 1 COW = 0.15 USDC
+  const usdcPrice = utils.parseUnits("0.15", 6);
+  // - 1 GNO = 400 USDC
+  const usdPerCow = ethers.utils.parseEther("0.15");
+  const gnoPrice = usdPerCow.div(400);
+  // - 1 WETH = 4000 USDC
+  const nativeTokenPrice = usdPerCow.div(4000);
+  const priceDenominator = ethers.utils.parseEther("1");
+  const wethBalanceOfUser = ethers.utils.parseUnits("234", 18);
+  let deploymentData: DeploymentData;
+  let vCowTokenSupply: BigNumber;
+  let provenClaims: ProvenClaims;
+  let claim: Claim;
+  let claims: ProvenClaim[];
+
+  it("User Option: claims the user option and vest it", async () => {
+    claim = {
+      account: user.address,
+      claimableAmount: ethers.utils.parseUnits("1234", 18),
+      type: ClaimType.UserOption,
+    };
+    provenClaims = computeProofs([claim]);
+    claims = provenClaims.claims;
+
+    const deploymentParameters: DeploymentParameters = {
+      deployer,
+      cowDao,
+      communityFundsTarget,
+      investorFundsTarget,
+      teamController,
+      provenClaims,
+      usdcPrice,
+      gnoPrice,
+      nativeTokenPrice,
+      usdPerCow,
+      initialCowSupply,
+    };
+    deploymentData = await standardDeployment(deploymentParameters);
+
+    const wethToPay = claim.claimableAmount
+      .mul(nativeTokenPrice)
+      .div(priceDenominator);
+    await deploymentData.wethToken
+      .connect(deployer)
+      .mint(user.address, wethBalanceOfUser);
```

Comment:
Since we use native ETH in the frontend, we might want to do that also here instead of creating a WETH token.
All the rest should be quite similar (except that it should not be the user address itself that sends the claim transaction, otherwise the eth amount also includes gas).

(After reading this PR fully, I see there is another test for claiming from another user. Maybe that test could use ETH instead?)